check/src/kindcheck.rs

```diff
@@ -73,15 +50,15 @@ impl<'a> KindCheck<'a> {
             subs: subs,
             type_kind: typ.clone(),
             function1_kind: function1_kind.clone(),
```

Comment:
Ohhh. But you could do `function2_kind: Kind::function(vec![typ.clone(), typ.clone()], typ)`. Not sure which is better...